web/src/main/java/com/es/phoneshop/web/controller/pages/CartPageController.java

```diff
@@ -1,25 +1,101 @@
+    private final CartService cartService;
+    private final Validator validator;
 
-    @RequestMapping(method = RequestMethod.GET)
-    public void getCart() {
-        cartService.getCart();
+    private static final String SUCCESS_MESSAGE = "Successful cart updating";
+
+    public CartPageController(CartService cartService, @Qualifier("cartDTOValidator") Validator validator) {
+        this.cartService = cartService;
+        this.validator = validator;
+    }
+
+    @ModelAttribute(value = "cartDTO")
+    public void addCartAttribute(Model model) {
+        model.addAttribute("cartDTO", getCartDTO());
+    }
+
+    @GetMapping
+    public String getCart(Model model) {
+        model.addAttribute("cart", cartService.getCart());
+
+        return "cart";
+    }
+
+    @PostMapping
+    public String updateCart(@ModelAttribute("cartDTO") CartDTO cartDTO, BindingResult bindingResult,
+                             RedirectAttributes redirectAttributes) throws OutOfStockException {
+        validator.validate(cartDTO, bindingResult);
+
+        if (bindingResult.hasErrors()) {
+            redirectAttributes.addFlashAttribute("cartDTO", cartDTO);
+            redirectAttributes.addFlashAttribute("errors", getErrors(bindingResult));
+            return "redirect:/cart";
+        }
+
+        cartService.update(mapCartDTOToMap(cartDTO));
+
+        redirectAttributes.addFlashAttribute("successMessage", SUCCESS_MESSAGE);
+        return "redirect:/cart";
+    }
+
+    @PostMapping("/delete/{phoneId}")
+    public String deleteCartItem(@PathVariable Long phoneId) {
+        cartService.remove(phoneId);
+
+        return "redirect:/cart";
+    }
+
+    private static Map<Long, Long> mapCartDTOToMap(CartDTO cartDTO) {
+        return new HashMap<>(cartDTO.getItems().stream()
+                .collect(Collectors.toMap(CartItemDTO::getPhoneId, CartItemDTO::getQuantity)));
+    }
+
+    private CartDTO getCartDTO() {
+        CartDTO cartDTO = new CartDTO();
+        cartDTO.setItems(cartService.getCart().getItems().stream()
+                .map(cartItem -> new CartItemDTO(cartItem.getPhone().getId(), (long) cartItem.getQuantity()))
+                .collect(Collectors.toList()));
+        return cartDTO;
     }
 
-    @RequestMapping(method = RequestMethod.PUT)
-    public void updateCart() {
-        cartService.update(null);
+    private Map<String, String> getErrors(BindingResult bindingResult) {
+        List<FieldError> fieldErrorList = bindingResult.getFieldErrors();
+        Map<String, String> errors = new LinkedHashMap<>();
+        for (FieldError fieldError : fieldErrorList) {
+            if (fieldError.getDefaultMessage().contains("java.lang.NumberFormatException")) {
+                errors.put(fieldError.getField(), "Quantity should be an integer");
+            } else {
+                errors.put(fieldError.getField(), fieldError.getDefaultMessage());
+            }
+        }
+        System.out.println(errors);
```

Comment:
Get the habit of cleaning your code from unnecessary logging before raising a PR :)